bubble_sort.rb

```diff
@@ -0,0 +1,90 @@
+#Method definition
+def bubble_sort(array)
+    # Get the length of the array
+    n = array.length
+    # If the size is equal to 1 return the array as it is already sorted
+    if n == 1
+        array
+    # If the size is 0 return a message saying the array is empty
+    elsif n == 0
+        return "Array is empty"
+        
+    else
+        # If the array has more than 1 element perform the following
+        # We define an loop that breaks when the whole array is sorted
+        loop { 
+            # We create a sorted variable and initialize it to false
+            sorted = false
+            # Itertate through the loop n-1 times
+            (n-1).times { |i|
+                # If the current array element (array[i]) is greater than the next one (array[i+1])
+                if array[i] > array[i+1]
+                    # swap the values
+                    array[i],array[i+1] = array[i+1],array[i]
+                    # update the sorted variable to true
+                    sorted = true
+                else
+                    # Otherwise if it is less than or equal to the next leave them as they are since they are sorted
+                    array[i] <= array[i+1]
+                        array[i], array[i+1] = array[i], array[i+1]
+                    
+                
+                    
+                end
+
+
+            }
+            # Break out of the loop if everything evaluates to true
+            break if !sorted
+        }
```

Comment:
There is a better way to do this loop (the "ruby way") You can create a while loop that runs until your sorted variable is true.